options/posix/include/pthread.h

```diff
@@ -93,6 +93,18 @@ struct  __mlibc_barrier {
 };
 typedef struct __mlibc_barrier pthread_barrier_t;
 
+struct __mlibc_fair_rwlock {
+	pthread_mutex_t m;
+	pthread_mutex_t rw;
+	unsigned read_count;
+};
+typedef struct __mlibc_fair_rwlock pthread_rwlock_t;
+
+struct __mlibc_rwlockattr {
+	int align;
```

Comment:
Where is this inspired from? musl?

Reply:
the `__mlibc_rwlockattr`? That was modified from linux iirc